wdl-lint/src/rules/call_input_spacing.rs

```diff
@@ -0,0 +1,171 @@
+                    let next = c.next_sibling_or_token().unwrap();
+                    match next.kind() {
+                        // The opening brace of a call block must be followed by exactly one space,
+                        // the input keyword, a colon and a newline
+                        SyntaxKind::Whitespace => {
+                            if next.to_string() != " " {
+                                // Check for a single space
+                                state.add(call_input_spacing(c.text_range().to_span()));
+                            } else {
+                                // Check for the input keyword
+                                match next.next_sibling_or_token().unwrap().kind() {
+                                    SyntaxKind::InputKeyword => {}
+                                    _ => {
+                                        state.add(call_input_spacing(c.text_range().to_span()));
+                                    }
+                                }
+                            }
+                        }
+                        _ => {
+                            // Opening brace is followed by something other than whitespace
+                            state.add(call_input_spacing(c.text_range().to_span()));
+                        }
+                    }
+                }
+                SyntaxKind::InputKeyword => {
+                    input_seen = true;
+                    newline_seen = 0;
+                }
+                SyntaxKind::Whitespace => {
+                    if c.to_string().contains('\n') {
+                        if !input_seen {
+                            // Newlines are not allowed before the input keyword
+                            state.add(call_input_keyword_preceding_newline(
+                                c.text_range().to_span(),
+                            ));
+                        } else {
+                            newline_seen += 1;
+                        }
+                    }
+                }
+                SyntaxKind::CallInputItemNode => {
+                    if newline_seen > 0 {
+                        // Reset newlines seen, since this is an input
+                        // Empty lines will be detected by the Whitespace rule
+                        newline_seen = 0;
+                    } else if inputs > 1 {
+                        // Only check for newlines if there are multiple inputs
+                        state.add(call_input_missing_newline(c.text_range().to_span()));
+                    }
+
+                    // Check for assignment spacing
+                    if c.to_string().contains('=') && !c.to_string().contains(" = ") {
+                        let i = c.to_string().find('=').unwrap();
+                        state.add(call_input_assignment(Span::new(
+                            c.text_range().to_span().start() + i,
+                            1,
+                        )));
+                    }
```

Comment:
Can this be moved into a visitor for `CallInputItemNode`s? This `CallStatement` visitor is doing a lot, would be helpful to spread things out a bit IMO.